### PR TITLE
`azurerm_virtual_desktop_host_pool` - add custom_rdp_property

### LINKED ISSUE #8800: Support for WVD Hostpool RDP Properties
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently it is possible to create Windows Virtual Desktop Hostpools through Terraform but not to configure any of the RDP properties such as blocking device redirection, clipboard access etc.
Adding this functionality would make life a lot easier as it would reduce my teams dependancy on ARM templates

Thanks for the great work you're all doing, keep it up!

### New or Affected Resource(s)

* azurerm_virtual_desktop_host_pool

### Potential Terraform Configuration

One possible implementation could be to provide the RDP configuration as a string which is how I've been doing it in ARM
```hcl
resource "azurerm_virtual_desktop_host_pool" "personalautomatic" {
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  name                             = "personalautomatic"
  friendly_name                    = "personalautomatic"
  description                      = "Acceptance Test: A Personal host pool - personalautomatic"
  type                             = "Personal"
  load_balancer_type               = "Persistent"
  personal_desktop_assignment_type = "Automatic"

  rdp_properties = "redirectclipboard:i:0;camerastoredirect:s:*;redirectprinters:i:0"

}
```
Or there could be an rdp_properties block that supports the various options
```hcl
resource "azurerm_virtual_desktop_host_pool" "personalautomatic" {
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  name                             = "personalautomatic"
  friendly_name                    = "personalautomatic"
  description                      = "Acceptance Test: A Personal host pool - personalautomatic"
  type                             = "Personal"
  load_balancer_type               = "Persistent"
  personal_desktop_assignment_type = "Automatic"

  rdp_properties {
    redirectclipboard = false
    camerastoredirect = "*"
    redirectprinters  = false
  }

}
```
### References

* https://docs.microsoft.com/en-us/windows-server/remote/remote-desktop-services/clients/rdp-files
* #3507 
* #8605

### PR DESCRIPTION
**Pull request by MattiasAng**
fixes #8800

### CODE REVIEW

File: azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go [hunk 1]
```diff
@@ -99,6 +99,11 @@ func resourceVirtualDesktopHostPool() *schema.Resource {
 				Default:  false,
 			},
 
+			"custom_rdp_property": {
```
Comment (katbyte) [inline azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go:102]: Could we use 
```suggestion
			"custom_rdp_properties": {
```
as property implies 1, and thd documentation here suggests it actually sends multiple (there might also be an argument to make it a typelist too but thats your call)

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the PR @MattiasAng - i just have one comment about the schema but otherwise this looks good!

File: azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go (hunk 1 above)
Comment (MattiasAng) [reply azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go:102]: Not even Microsoft is bothered to do any kind of API for every separate setting, they just have a javascript that translates checkboxes to rdp properties string. So in short, this is good enough I think. :) 

Agreed on schema, was mostly following API naming.

File: azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go [hunk 2]
```diff
@@ -254,6 +260,7 @@ func resourceVirtualDesktopHostPoolRead(d *schema.ResourceData, meta interface{}
 		d.Set("preferred_app_group_type", string(props.PreferredAppGroupType))
 		d.Set("type", string(props.HostPoolType))
 		d.Set("validate_environment", props.ValidationEnvironment)
+		d.Set("custom_rdp_property", props.CustomRdpProperty)
```
Comment (katbyte) [inline azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource.go:263]:
```suggestion
		d.Set("custom_rdp_properties", props.CustomRdpProperty)
```

File: azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource_test.go [hunk 3]
```diff
@@ -152,6 +152,7 @@ resource "azurerm_virtual_desktop_host_pool" "test" {
   load_balancer_type       = "BreadthFirst"
   maximum_sessions_allowed = 100
   preferred_app_group_type = "Desktop"
+  custom_rdp_property      = "audiocapturemode:i:1;audiomode:i:0;"
```
Comment (katbyte) [inline azurerm/internal/services/desktopvirtualization/virtual_desktop_host_pool_resource_test.go:155]:
```suggestion
  custom_rdp_properties    = "audiocapturemode:i:1;audiomode:i:0;"
```

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @MattiasAng - missed a couple but i'll fix that up

File: website/docs/r/virtual_desktop_host_pool.html.markdown [hunk 4]
```diff
@@ -25,6 +25,7 @@ resource "azurerm_virtual_desktop_host_pool" "example" {
   name                     = "pooleddepthfirst"
   friendly_name            = "pooleddepthfirst"
   validate_environment     = true
+  custom_rdp_properties      = "audiocapturemode:i:1;audiomode:i:0;"
```
Comment (katbyte) [inline website/docs/r/virtual_desktop_host_pool.html.markdown:28]:
```suggestion
  custom_rdp_properties    = "audiocapturemode:i:1;audiomode:i:0;"
```

**Review by katbyte — APPROVED** [review]

**Pull request merged**